contracts/governance/src/IZeroExVotes.sol

```diff
@@ -23,6 +23,12 @@ interface IZeroExVotes {
     struct Checkpoint {
         uint32 fromBlock;
         uint224 votes;
+        uint224 quadraticVotes;
```

Comment:
This should be possible to pack this into a single struct as
```Solidity
struct Checkpoint {
    uint8 version;
    uint40 fromBlock; // best to use 40 bits to avoid overflow on all reasonable timescales
    uint96 votes; // actually only 90 bits
    uint48 quadraticVotes; // actually only 45 bits
}
```
Which also gives 64 bits left over for future work. The `version` tag makes future upgrades easier. I also recommend setting the initial version tag to 1, not 0, to make it easier to detect uninitialized structs.

Reply:
Atm these are just the OZ types used I haven't storage optimised anywhere. I agree with the storage size for `votes` and `quadraticVotes` but I don't see why is adding a `version` field needed, considering only storage non-breaking changes will be added, e.g. at some point we might add a `uint64` to the struct. What problem do you see with that?

Reply:
Setting a `version` makes it much easier to detect whether some future, hypothetical additional `uint64` field is semantically zero or unallocated/uninitialized (and should therefore be initialized to some nonzero value). It's probably possible in all cases to be clever about it and use the `fromBlock` field for the same purpose (assuming we know which block a backwards-incompatible change was made). This is a weak recommendation.

Reply:
I prefer the data normalisation of your weak recommendation :) Just to step through the example if at block X we upgrade from V1 to V2, the voting logic as it stands is
```
if (checkpoint exists for block X) {
            // Update checkpoint
        } else {
            // Push new checkpoint
        }
```
This means that for blocks after the one of the upgrade tx, new compatible `Checkpoint`s will be pushed. For any checkpoint updates that happen in the _same_ block and _before_ the upgrade tx will succeed whereas the ones _after_ will succeed but only partially update the Checkpoint excluding any new property sets, i.e. they's behave like the previous version for that one block.

Reply:
For posterity just sharing what we agreed offline that is to not have versioning in the Checkpoint or otherwise as this can be safely added with an upgrade in the case where backward incompatible vote calculation logic is ever added, i.e. vote weight calculation that requires historical Checkpoint data.